rpc/cloudagent/compute/baremetalhost/moc_cloudagent_baremetalhost.proto

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache v2.0 license.
+
+syntax = "proto3";
+option go_package = "github.com/microsoft/moc/rpc/cloudagent/compute";
+package moc.cloudagent.compute;
+
+import "google/protobuf/wrappers.proto";
+import "moc_common_common.proto";
+import "moc_cloudagent_virtualmachine.proto";
+
+enum BareMetalHostPowerState {
+	Unknown = 0;
+	Running = 1 ;
+	Off = 2;
+}
+
+message BareMetalHostRequest {
+	repeated BareMetalHost BareMetalHosts = 1;
+	Operation OperationType = 2;
+}
+
+message BareMetalHostResponse {
+	repeated BareMetalHost BareMetalHosts = 1;
+	google.protobuf.BoolValue Result = 2;
+	string  Error = 3;
+}
+
+message BareMetalHostDisk {
+	string diskName = 1;
+	int32 diskSizeGB = 2;
+}
+
+message BareMetalHostStorageConfiguration {
+	repeated BareMetalHostDisk disks = 1;
+}
+
+message BareMetalHostNetworkInterface {
+	string networkInterfaceName = 1;
+}
+
+message BareMetalHostNetworkConfiguration {
+	repeated BareMetalHostNetworkInterface interfaces = 1;
+}
+
+message BareMetalHostSize {
+	int32 cpuCount = 1;
+	int32 gpuCount = 2;
+	int32 memoryMB = 3;
+}
+
+message BareMetalHostHardwareConfiguration {
+	BareMetalHostSize machineSize = 1;
+}
+
+message BareMetalHost {
+	string name = 1;
+	string id = 2;
+	BareMetalHostStorageConfiguration storage = 3;
+	BareMetalHostNetworkConfiguration network = 4;
+	BareMetalHostHardwareConfiguration hardware = 5;
+	BareMetalHostPowerState powerState = 6;
+	SecurityConfiguration security = 7;
+	string bareMetalMachineName = 8;
```

Comment:
is this the hostName? can you keep this as MachineName. BareMetal is redundant

Reply:
or is this reference to the bareMetalMachine - which doesnt seem right - to have this cyclic dependency.

You can maintain these in the internal structures, when machine gets added to host via AddResource

Reply:
This is a reference to `bareMetalMachine`. It is a read-only field that is populated by MOC for convenience and debugging.

Reply:
There are patterns to do this - look at how group manages the dependencies in groupsInternal structure

Lets follow the same pattern.

Reply:
We have the references in both Internal objects: https://github.com/microsoft/wssdcloudagent/blob/04751dc1fc85c66b5668d7148ae6cb5b455824d1/pkg/cloudagent/services/compute/baremetalhost/internal/baremetalhost.go#L16 and https://github.com/microsoft/wssdcloudagent/blob/04751dc1fc85c66b5668d7148ae6cb5b455824d1/pkg/cloudagent/services/compute/baremetalmachine/internal/baremetalmachine.go#L16. 

With that, do you suggest to delete references from both MOC objects (BareMetalHost and BareMetalMachine) and just keep both references in their Internal objects or are you suggesting something else?

Reply:
reference is required in proto, only if it needs to be input by the client.
For debugging purpose and other internal notification purpose, internal object implementation is good.

Unless required, do not expose properties in proto.

Reply:
With that, should any discovered information about BMH be only part of the internal object? Say, CPU count.

Reply:
https://github.com/microsoft/moc/blob/main/rpc/cloudagent/cloud/node/moc_cloudagent_node.proto#L39
https://github.com/microsoft/moc/blob/main/rpc/cloudagent/storage/container/moc_cloudagent_container.proto#L37

some sample code on how to propagate node info that is discovered

Reply:
Re info, I think I want to leave this to another time. I think as we add inspection support, these will need to be modified as well and so would rather do it in one go.